my-react-app/src/redux/game/reducer.js

```diff
@@ -0,0 +1,43 @@
+import calculateWinner from '../../app/screens/Game/utils';
+
+import { actions } from './actions';
+
+const initialState = {
+  history: [{ squares: Array(9).fill(null) }],
+  stepNumber: 0,
+  xIsNext: true
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case actions.handleClick: {
+      const history = state.history.slice(0, state.stepNumber + 1);
+      const current = history[history.length - 1];
+      const squares = current.squares.slice();
+      if (calculateWinner(squares) || squares[action.pos]) {
```

Comment:
This logic shouldn't be here. In the component where you click the square, check if you should dispatch the action or not.